### PR TITLE
Wire MEDIA scalar through server + client UI

### LINKED ISSUE #418: Item schema type for media type unification
Currently you can only have `Array > Images` or `Array > Videos` or `Array > Audio` etc.
It'd be much more convient to have `Array > Media`. This would automatically translate the URL of the media to image/video/Audio
It should also be able to be used for `Map > Media` or just `Media`

### PR DESCRIPTION
**Pull request by julietshen**
Follow-up to #605. v2.3.0 of @roostorg/coop-types is published — both packages bump to `^2.3.0`. Closes #418.

## What this delivers
- MEDIA appears in the ItemType form field-type dropdown automatically, since `ItemTypeFormCustomField` iterates `Object.values(GQLScalarType)`. Adopters can now declare a `Media` / `Array<Media>` / `Map<key, Media>` field at item-type creation time.
- The server coerces a MEDIA submission `"https://…"` into `{ url, mediaType: 'AUDIO' | 'IMAGE' | 'VIDEO' | null }`, with `mediaType` resolved from the URL pathname extension at ingestion. The detection table is conservative — ambiguous containers like `.ogg` stay `null` so a downstream consumer can probe Content-Type instead.
- `ManualReviewJobFieldsComponent` renders a MEDIA value as image / video / audio per the stored `mediaType`, falling back to a plain link when detection didn't resolve. Existing AUDIO/IMAGE/VIDEO render paths are unchanged.

## Changes
**Server**
- `server/graphql/schema.ts` — adds `MEDIA` to `ScalarType`, `FieldType`, `SignalInputType`.
- `server/services/itemProcessingService/fieldTypeHandlers.ts` — new `coerceMediaInput` handler; exports `detectMediaKindFromUrl` for reuse.
- `server/services/itemProcessingService/fieldTypeHandlers.test.ts` — +29 unit tests (URL→kind happy paths, unresolved cases, scheme/type rejection, query-string edge case).
- `server/test/arbitraries/ContentType.ts` — `MediaArbitrary` entry so property tests over `ScalarType` don't break.
- `server/graphql/generated.ts` + `client/src/graphql/generated.ts` — codegen regen.

**Client**
- `ManualReviewJobFieldsComponent.tsx` — new MEDIA render case that dispatches by `mediaType`. Adds MEDIA to the three exhaustive helper switches (scalar-passthrough, `valueCanBeStringified`, scalar-shouldn't-hit-container).
- `RuleTestModal.tsx`, `RuleInsightsSamplesTable.tsx`, `models/signal.ts`, `dashboard/rules/types.ts`, `item_types/itemTypeUtils.ts` — propagate MEDIA through the other exhaustive `ScalarType` switches and the rules-side `isMediaType` helper.

**Dependencies**
- `server/package.json`, `client/package.json` — `@roostorg/coop-types ^2.2.0 → ^2.3.0`.

## AGENTS.md callouts
- The `satisfies FieldType` guard at `server/graphql/modules/itemType.ts:1053` — this PR satisfies that guard since the server's `FieldType` (from coop-types 2.3.0) now includes MEDIA.
- New scalar GQL enum value: additive, no Apollo cache risk.

## Test plan
- [x] `cd server && npm run build` — clean
- [x] `cd server && npx jest fieldTypeHandlers` — 33 tests pass
- [x] `cd server && npm run lint` — 0 errors (pre-existing warnings only)
- [x] `cd client && npm run build` — clean
- [x] `cd client && npm run lint` — only pre-existing `itemTypeCodeSampleUtils.ts` JsonValue errors (same on main)
- [ ] CI green
- [ ] Reviewer spot-checks the MEDIA render branch in the MRT against a real `Array<MEDIA>` item

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added first-class MEDIA field type across UI, rule testing, insights, and ingestion paths.
  * Manual review now displays media (image/video/audio) with appropriate viewers and fallbacks.
  * Placeholders and sample values for MEDIA included in rule testing.

* **Validation & Detection**
  * Automatic media-type detection from URLs and coercion into {url, mediaType} forms.

* **Tests & Chores**
  * Expanded tests for MEDIA coercion/detection and dependency version bumps for client/server.

[Review Change Stack](https://app.coderabbit.ai/change-stack/roostorg/coop/pull/632?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
     "@radix-ui/react-slider": "^1.2.0",
     "@radix-ui/react-switch": "^1.1.0",
     "@radix-ui/react-tooltip": "^1.1.2",
-    "@roostorg/coop-types": "^2.2.0",
+    "@roostorg/coop-types": "^2.3.0",
```
[the remark this answers is not in the document]
Comment (julietshen) [reply client/package.json:39]: this was done by a human, resolving

File: server/package.json [hunk 2]
```diff
@@ -40,7 +40,7 @@
     "@opentelemetry/api": "^1.8.0",
     "@opentelemetry/semantic-conventions": "^1.22.0",
     "@roostorg/coop-integration-example": "^2.0.0",
-    "@roostorg/coop-types": "^2.2.0",
+    "@roostorg/coop-types": "^2.3.0",
```
[the remark this answers is not in the document]
Comment (julietshen) [reply server/package.json:43]: same comment as above, i did this manually

**Comment by julietshen** [comment]
Screenshot 2026-05-28 at 3 47 40 PM

Screenshot 2026-05-28 at 3 47 44 PM

tested locally and Media now shows up

**Comment by julietshen** [comment]
validated by an adopter!

**Review by ThatKoffe — APPROVED** [review]

**Review by juanmrad — APPROVED** [review]
LGTM. code is sound.
